### PR TITLE
new package: netsurf 3.9

### PR DESCRIPTION
**Pull request by a-schaefers**
## Checklist

- [x] Latest upstream version.
- [x] I agree to maintain this package (**required**).
- [x] I agree to notify KISS if I can no longer maintain this package (**required**).

### CODE REVIEW

**Comment by a-schaefers** [comment]
i don't understand how to make vim only a build time dep, please help me fix this

File: community/netsurf/depends [hunk 1]
```diff
@@ -0,0 +1,6 @@
+curl
+gtk+2
+vim vim
```
Comment (dylanaraps) [inline community/netsurf/depends]: `vim make`

Adding `make` after a dependency specifies it as a build and not runtime dependency. :)

File: community/netsurf/sources [hunk 2]
```diff
@@ -0,0 +1 @@
+http://download.netsurf-browser.org/netsurf/releases/source-full/netsurf-all-3.9.tar.gz
```
Comment (dylanaraps) [inline community/netsurf/sources]: Any `https` source available?

File: community/netsurf/depends (hunk 1 above)
Comment (a-schaefers) [reply community/netsurf/depends]: oh finally get it now. I read the website section 3 or 4 times, too, maybe it should be reworded, or maybe I need more sleep.

File: community/netsurf/build [hunk 3]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh -e
+
+make
+make DESTDIR="$1" install
```
Comment (dylanaraps) [inline community/netsurf/build]: No need to set `--prefix=/usr` with `./configure`?

**Comment by dylanaraps** [comment]
Interesting that `vim` is a make dependency! Know why?

File: community/netsurf/build (hunk 3 above)
Comment (a-schaefers) [reply community/netsurf/build]: Correct. This netsurf is very purist in the way they do things, if you haven't tried it btw, you might be impressed!

Comment (a-schaefers) [reply community/netsurf/build]: actually wait now im confused too

Comment (a-schaefers) [reply community/netsurf/build]: i have it installed, looks like it's in /usr/bin but i don't know which route it took, e.g. did it get installed through the symlink or proper?

**Comment by a-schaefers** [comment]
> Interesting that `vim` is a make dependency! Know why?

Nooo  idea, it's weird. It uses the "xss" of vim iirc...

File: community/netsurf/build (hunk 3 above)
Comment (dylanaraps) [reply community/netsurf/build]:
```sh
make \
    LIBDIR=lib \
    INCLUDEDIR=include \
    PREFIX=/usr

make \
    LIBDIR=lib \
    INCLUDEDIR=include \
    PREFIX=/usr \
    install
```

Comment (dylanaraps) [reply community/netsurf/build]: Also, enable GTK3 as GTK2 will be dropped from KISS once Firefox drops it.

```
    TARGET=gtk \
    NETSURF_GTK_MAJOR=3 \
```

Comment (a-schaefers) [reply community/netsurf/build]: okay, thought gtk2 might be more forward looking (lol)

Comment (a-schaefers) [reply community/netsurf/build]: @dylanaraps what about DESTDIR

**Comment by dylanaraps** [comment]
Builds and works fine? Do we need to do anything special to get working SSL support?

**Comment by a-schaefers** [comment]
I typed kiss b netsurf and it failed build because didn't pull in gtk+3 to build first ends/gtk/Makefile:46: *** Unable to find library for: GTK-2 (gtk+-2.0).  Stop.
make

**Comment by dylanaraps** [comment]
My bad. Target should be: `TARGET=gtk3`. :+1:

**Comment by a-schaefers** [comment]
give me a bit to compile gtk+3 before i can test this, because I just purged it a second ago, and realized now i have no cached pkg of it

**Comment by konimex** [comment]
`vim` provides `xxd`. Fortunately it's a `make` depends.

**Comment by a-schaefers** [comment]
> `vim` provides `xxd`. Fortunately it's a `make` depends.

No kidding from a guy who uses Emacs as his login shell in /etc/passwd

**Comment by a-schaefers** [comment]
build failed again https://termbin.com/scfj

**Comment by dylanaraps** [comment]
Arch sets:

```sh
   NETSURF_GTK_MAJOR=3 \
    TARGET=gtk \```

For both `make` and `make install`.

**Comment by a-schaefers** [comment]
> Arch sets:
> 
> ```shell
>    NETSURF_GTK_MAJOR=3 \
>     TARGET=gtk \
> ```
> 
> For both `make` and `make install`.

ok but if you look at the output i do not believe it made it to the second make.

**Comment by a-schaefers** [comment]
okay now it has a new error, this is good https://termbin.com/7k99

**Comment by dylanaraps** [comment]
```
 COMPILE: build/Linux-gtk3/messages_gresource.c
    LINK: nsgtk3
```

It's using GTK3 now at least.

Looks like a simple `sed` command may be needed to fix the `install` call.

**Comment by a-schaefers** [comment]
I think I got it now

**Comment by a-schaefers** [comment]
A missing slash before install, you see it?

**Comment by dylanaraps** [comment]
I do. :)

**Comment by a-schaefers** [comment]
Gtk3 is ugly man i lose that classy look

**Comment by a-schaefers** [comment]
Do we even have any themes or icons in the repos

**Comment by a-schaefers** [comment]
build success ready for merge

**Comment by dylanaraps** [comment]
We do not. I personally don't use any themes so I haven't packaged any.

**Comment by a-schaefers** [comment]
i dont use any either, gtk2 looks better tho imho ;)

**Pull request merged**